jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -457,6 +470,56 @@ ecma_utf8_string_to_number (const lit_utf8_byte_t *str_p, /**< utf-8 string */
     return ecma_number_make_infinity (sign);
   }
 
+  if (!is_infinity_check)
+  {
+    lit_utf8_size_t e_position = 0;
+    lit_utf8_size_t dot_position = 0;
+
+    while (e_position < str_size)
+    {
+      if (str_p[e_position] == LIT_CHAR_DOT)
+      {
+        dot_position = e_position;
+      }
+      else if (str_p[e_position] == LIT_CHAR_LOWERCASE_E || str_p[e_position] == LIT_CHAR_UPPERCASE_E)
+      {
+        break;
+      }
+
+      e_position++;
+    }
+
+    if (e_position != str_size && e_position != 0)
+    {
+      ecma_number_t base = ecma_utf8_string_to_number (str_p, e_position, true);
+      ecma_number_t exponent = ecma_utf8_string_to_number (str_p + e_position + 1, str_size - e_position - 1, true);
```

Comment:
I didn't have the time to get my head around the whole code, yet, but this part looks suspicious. 1) This function was non-recursive and now just became recursive. Is that really needed? 2) base and exponent are of ecma_number_t type, which is a floating point type but then we use the variables as integers (roughly). But what if the recursively called ecma_utf8_string_to_number returns INF, -INF, NAN, or something non-integer in general? If that's guaranteed not to happen, isn't there another function that could return something integer? (If there isn't, shouldn't we write one instead of re(ab)using ecma_utf8_string_to_number?

(Note: as stated above, these are just suspicions, no specific requests based on deep understanding of the code.)